src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -28,188 +29,84 @@
-            "TemplateSizeAnomaly",
-            "MateUnmapped", "InterContigPair",
+    private static final boolean USE_FAST_MATH_EXP = true;
+
+    private static final List<String> DEFAULT_EVIDENCE_TYPE_ORDER = Arrays.asList(
+            "TemplateSizeAnomaly", "MateUnmapped", "InterContigPair",
             "SplitRead", "LargeIndel", "WeirdTemplateSize", "SameStrandPair", "OutiesPair"
     );
-
-    static private final int NUM_OVERLAP = 0;
-    static private final int NUM_COHERENT = 1;
-    static private final int OVERLAP_MAPPING_QUALITY = 2;
-    static private final int COHERENT_MAPPING_QUALITY = 3;
+    private static final Map<String, Integer> evidenceTypeMap = evidenceTypeOrderToImmutableMap(DEFAULT_EVIDENCE_TYPE_ORDER);
+    private static final String DEFAULT_PREDICTOR_RESOURCE_PATH = "/large/sv_evidence_classifier.bin";
 
     private final PartitionCrossingChecker partitionCrossingChecker;
 
     private final Predictor predictor;
-    private final Map<String, Integer> evidenceTypeMap;
-    private final double coverage;
     private final double thresholdProbability;
-    private final Map<String, String> readGroupToLibraryMap;
-    private final Map<String, LibraryStatistics> libraryStatisticsMap;
+    private final ReadMetadata readMetadata;
 
     private final EvidenceOverlapChecker evidenceOverlapChecker;
-    private final Map<BreakpointEvidence, int[]> rawFeatureCache;
+    private final Map<BreakpointEvidence, UnscaledOverlapInfo> rawFeatureCache;
 
     private Iterator<SVIntervalTree.Entry<List<BreakpointEvidence>>> treeItr;
     private Iterator<BreakpointEvidence> listItr;
-    private final SVIntervalTree<Double> genomeGaps;
-    private final SVIntervalTree<Double> umapS100Mappability;
+    private final FeatureDataSource<BEDFeature> genomeGaps;
+    private final FeatureDataSource<BEDFeature> umapS100Mappability;
 
-    public XGBoostEvidenceFilter(
+    XGBoostEvidenceFilter(
             final Iterator<BreakpointEvidence> evidenceItr,
             final ReadMetadata readMetadata,
             final StructuralVariationDiscoveryArgumentCollection.FindBreakpointEvidenceSparkArgumentCollection params,
             final PartitionCrossingChecker partitionCrossingChecker
     ) {
-        this.predictor = loadPredictor(params.svEvidenceFilterModelFile);
-        this.evidenceTypeMap = loadEvidenceTypeMap(params.svCategoricalVariablesFile);
-        this.genomeGaps = new SVIntervalTree<>();
-        loadGenomeIntervals(this.genomeGaps, params.svGenomeGapsFile, readMetadata);
-        loadGenomeIntervals(this.genomeGaps, params.svGenomeCentromeresFile, readMetadata);
-        this.umapS100Mappability = new SVIntervalTree<>();
-        loadGenomeIntervals(this.umapS100Mappability, params.svGenomeUmapS100File, readMetadata);
+        if(params.svGenomeGapsFile == null) {
+            throw new IllegalArgumentException("XGBoostEvidenceFilter requires non-null sv-genome-gaps-file");
```

Comment:
If you don't have a gaps file, will the classifier not be able to run, or will it just have "0" on the "gap" feature for every piece of evidence? Just wondering if this should be a hard rule.

Reply:
It's easy to provide a trivial gaps file (empty file with extension .bed.gz _should_ work), so it seemed like a better idea to ensure they deliberately opted out rather than forgot to specify. Especially since adding gaps helps a lot (3rd place in feature importance). Of course I could set gaps to 0 like you said, and the classifier would do its best assuming you're in a gap-free region.

Reply:
Or is there some other clean way to have them specify "no" to the gaps file. i.e. 3 possible states: default (passed nothing), opt out (pass ???), valid file name. I'm not experienced with java arguments, so I don't know the idioms.

Reply:
Forcing them to make an empty bed file seems like makework. What if you made an additional optional flag `--runWithoutGapsAnnotation` or something like that, and verify that they either provided a value for the file or set the flag?

Reply:
That makes sense. I'll make that change and update documentation. I'm thinking I'll need an integration test for that case as well?

Reply:
Sorry, one more question. What should happen if they set the flag _and_ pass a valid gaps file? It seems like it's best to use the gaps file in that case, maybe throw a warning?

Reply:
I think I'd throw an `IllegalArgumentException` there, make them remove one or the other and try not to proceed with nonsensical arguments.

Reply:
Ok, I'll do that then.